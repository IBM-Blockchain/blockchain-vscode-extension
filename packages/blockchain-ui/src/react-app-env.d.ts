/// <reference types="react-scripts" />

interface File {
    path: string;
}
