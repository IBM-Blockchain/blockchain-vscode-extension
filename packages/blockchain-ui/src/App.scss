@import '@carbon/colors/scss/colors';

#root{
    .app-container__tutorial-page {
        background-color: $gray-10;
        position: absolute;
        width: 100%;
        z-index: -1;
        box-shadow: inset 0px 164x #FFFFFF;
        top: 0;
        left: 0;

    }
}
