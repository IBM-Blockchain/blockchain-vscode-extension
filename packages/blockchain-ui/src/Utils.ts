import IVSCode from './interfaces/IVSCode';
declare const vscode: IVSCode;

class Utils {

    static changeRoute(newPath: string): void {
        dispatchEvent(new MessageEvent('message', {
            data: {
                path: newPath
            }
        }));
    }

    static postToVSCode(message: { command: string, data?: any }): void {

        if (!(window as any)['Cypress']) {
            vscode.postMessage(message);
        } else {
            window.postMessage(message, '*');
        }
    }

}

export default Utils;
