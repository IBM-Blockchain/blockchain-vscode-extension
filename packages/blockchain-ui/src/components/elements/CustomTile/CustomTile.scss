@import '@carbon/layout/scss/_spacing';

.custom-tile {
    width: 90%;

    p {
        margin-bottom: $spacing-05;
        width: 80%
    }
}

.new-tab-img {
    float: right;
    width: 24px;
    height: 24px;

    &:hover {
        cursor: pointer;
    }
}

