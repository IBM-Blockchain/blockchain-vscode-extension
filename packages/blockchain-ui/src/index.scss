@import '@carbon/colors/scss/colors';
@import '@carbon/themes/scss/themes';
@import '@carbon/type/scss/type';
@import '@carbon/type/scss/font-face/_mono';
@import '@carbon/type/scss/font-face/_sans';
@import './mixins/theme-mixins.scss';

@include carbon--type-reset();
@include carbon--font-face-mono();
@include carbon--font-face-sans();

html {

    .vscode-dark {
        @include carbon--theme($carbon--theme--g90) {

            #root {
                // Color will default to #000000 otherwise
                color: $text-01;        
                @include select-dark;
            }

            // text inputs on the create transaction screen
            #create-txn-form {
                @include text-inputs-dark;
            }

            // data tables on home page
            #recent-txns-table, #saved-txns-table {
                @include data-table-dark;
            }

            // InlineSelect components
            #smart-contract-select-wrapper {
                @include inline-select-dark;
            }

            // Checkbox
            @include checkbox-dark;

            //panel containing transaction output
            #output-panel {
                @include output-panel-dark;
            }
        }
    }

    .vscode-light {
        @include carbon--theme($carbon--theme--g10) {
            
            #root {
                color: $text-01;
            }
        }
    }
}
