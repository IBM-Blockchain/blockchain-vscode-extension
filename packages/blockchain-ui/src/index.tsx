import * as React from 'react';
import * as ReactDOM from 'react-dom';
import App from './App';
import './index.scss';

const app: any = ReactDOM.render(
    <App/>,
    document.getElementById('root')
);

// @ts-ignore
if (window.Cypress) {
    // @ts-ignore
    window.app = app;
}
