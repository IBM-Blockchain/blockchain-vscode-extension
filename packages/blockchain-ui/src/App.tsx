import * as React from 'react';
import './App.scss';
import { HashRouter as Router, Route, Redirect } from 'react-router-dom';
import TransactionHome from './components/TransactionHome/TransactionHome';
import TransactionCreate from './components/TransactionCreate/TransactionCreate';
import ISmartContract from './interfaces/ISmartContract';
import Utils from './Utils';

interface AppState {
    redirectPath: string;
    gatewayName: string;
    activeSmartContract: ISmartContract;
    smartContracts: Array<ISmartContract>;
    transactionOutput: string;
}

class App extends React.Component<{}, AppState> {
    constructor(props: {}) {
        super(props);
        this.state = {
            redirectPath: '',
            gatewayName: '',
            activeSmartContract: {
                name: '',
                version: '',
                channel: '',
                label: '',
                transactions: [],
                namespace: ''
            },
            smartContracts : [],
            transactionOutput: ''
        };
        this.switchSmartContract = this.switchSmartContract.bind(this);
        this.postMessageHandler = this.postMessageHandler.bind(this);
    }

    componentDidMount(): void {
        window.addEventListener('message', (event: MessageEvent) => {
            if (event.data.output) {
                this.setState({
                    transactionOutput: event.data.output
                });
            } else {
                this.setState({
                    redirectPath: event.data.path,
                    gatewayName: event.data.state ? event.data.state.gatewayName : this.state.gatewayName,
                    activeSmartContract: event.data.state ? event.data.state.activeSmartContract : this.state.activeSmartContract,
                    smartContracts: event.data.state ? event.data.state.smartContracts : this.state.smartContracts,
                });
            }
        });
    }

    switchSmartContract(newActiveContractLabel: string): void {
        const smartContracts: Array<ISmartContract> = this.state.smartContracts;
        this.setState({
            activeSmartContract: smartContracts.find((obj: ISmartContract) => obj.label === newActiveContractLabel) as ISmartContract,
            smartContracts: smartContracts
        });
    }

    postMessageHandler(command: string, data?: any): void {
        if (data === undefined) {
            data = {
                gatewayName: this.state.gatewayName,
                activeSmartContract: this.state.activeSmartContract,
                smartContracts: this.state.smartContracts
            };
        }

        Utils.postToVSCode({
            command: command,
            data: data
        });
    }

    render(): JSX.Element {
        if (this.state.redirectPath === '') {
            return <div></div>;
        } else {
            return (
                <Router>
                    <div>
                        <Route render={(): JSX.Element => <Redirect push to={this.state.redirectPath}/>}></Route>
                        <Route exact path='/transaction' render={(): JSX.Element =>
                            <TransactionHome gatewayName={this.state.gatewayName} activeSmartContract={this.state.activeSmartContract} smartContracts={this.state.smartContracts}
                                switchSmartContract={this.switchSmartContract} postMessageHandler={this.postMessageHandler}/>}>
                        </Route>
                        <Route exact path='/transaction/create' render={(): JSX.Element =>
                            <TransactionCreate activeSmartContract={this.state.activeSmartContract} transactionOutput={this.state.transactionOutput}
                                postMessageHandler={this.postMessageHandler}/>}>
                        </Route>
                    </div>
                </Router>
            );
        }
    }
}

export default App;
