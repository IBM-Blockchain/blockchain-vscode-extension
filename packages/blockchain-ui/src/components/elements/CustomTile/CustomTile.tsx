import React, { Component } from 'react';
import { Tile } from 'carbon-components-react';
import HeadingCombo from '../HeadingCombo/HeadingCombo';
import { ExtensionCommands } from '../../../ExtensionCommands';
import Utils from '../../../Utils';
import newTabImg from '../../../resources/new-tab.svg';
import './CustomTile.scss';

interface IProps {
    title: string;
    body: string;
}

class CustomTile extends Component <IProps> {
    constructor(props: Readonly<IProps>) {
        super(props);

        this.tileClickHandler = this.tileClickHandler.bind(this);
    }

    tileClickHandler(): void {
        Utils.postToVSCode({
            command: ExtensionCommands.OPEN_TUTORIAL_GALLERY
        });
    }

    render(): JSX.Element {
        return (
            <Tile className='custom-tile' onClick={this.tileClickHandler}>
                <HeadingCombo
                    headingText={this.props.title}
                    subheadingText={this.props.body}
                />
                <img src={newTabImg} alt='' className='new-tab-img'/>
            </Tile>
        );
    }
}

export default CustomTile;
