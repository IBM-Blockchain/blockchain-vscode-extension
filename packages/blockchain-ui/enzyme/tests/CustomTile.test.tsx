import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import chai from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';

import CustomTile from '../../src/components/elements/CustomTile/CustomTile';
import Utils from '../../src/Utils';
import { ExtensionCommands } from '../../src/ExtensionCommands';

chai.should();
chai.use(sinonChai);

describe('CustomTile component', () => {
    let mySandBox: sinon.SinonSandbox;
    let postToVSCodeStub: sinon.SinonStub;

    beforeEach(() => {
        mySandBox = sinon.createSandbox();
        postToVSCodeStub = mySandBox.stub(Utils, 'postToVSCode').resolves();
    });

    afterEach(() => {
        mySandBox.restore();
    });

    it('should render the expected snapshot', () => {
        const component: any = renderer
            .create(<CustomTile title='My Tile' body='Some text I want to display in my tile'/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should post to VS Code when the new tab button is clicked', () => {
        const component: any = mount(<CustomTile title='My Tile' body='Some text I want to display in my tile'/>);
        component.find('img').simulate('click');
        postToVSCodeStub.should.have.been.calledOnceWithExactly({
            command: ExtensionCommands.OPEN_TUTORIAL_GALLERY
        });
    });
});
