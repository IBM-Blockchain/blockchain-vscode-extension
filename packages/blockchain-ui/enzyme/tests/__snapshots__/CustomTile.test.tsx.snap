// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTile component should render the expected snapshot 1`] = `
<div
  className="bx--tile custom-tile"
>
  <div
    className="heading-combo-container"
  >
    <h3>
      My Tile
    </h3>
    <p>
      Some text I want to display in my tile
    </p>
  </div>
  <img
    alt=""
    className="new-tab-img"
    onClick={[Function]}
    src="test-file-stub"
  />
</div>
`;
